docs/express-motivation.qmd

```diff
@@ -0,0 +1,343 @@
+
+## Decoupling {#decoupling}
+
+A big reason for Express' simplicity is that things like render functions get displayed exactly where they appear in the code.
+This is great for basic apps, since you don't have to separate backend from frontend (aka server from UI) logic.
+However, this can lead to code that's harder to reason about when it grows in size and complexity.
+For example, consider this snippet Express app -- can you visualize, at a glance, how the overall UI is structured?
+
+<details open>
+  <summary style="margin-bottom: -1em;">Hide code</summary>
+```{.python}
+with ui.layout_columns():
+    with ui.card():
+        @render.plot
+        def pair_plot():
+            df = load_penguins()
+            if df is None:
+                print("Dataframe is empty")
+                return
+
+            # Drop rows with missing values
+            df = df.dropna()
+
+            # Get list of features
+            features = df.select_dtypes(include=[np.number]).columns.tolist()
+
+            # Create a figure and axes with a subplot for each pair of features
+            fig, axs = plt.subplots(len(features), len(features), figsize=(15, 15))
+
+            # Create scatter plots for each pair of features
+            for i in range(len(features)):
+                for j in range(len(features)):
+                    if i != j:
+                        for species in df['species'].unique():
+                            axs[i, j].scatter(df[df['species']==species][features[i]],
+                                              df[df['species']==species][features[j]],
+                                              label=species)
+                        axs[i, j].set_xlabel(features[i])
+                        axs[i, j].set_ylabel(features[j])
+                    else:
+                        axs[i, j].text(0.5, 0.5, features[i], ha='center', va='center')
+
+            # Add a legend
+            handles, labels = axs[0, 1].get_legend_handles_labels()
+            fig.legend(handles, labels, loc='upper center')
+
+            fig.tight_layout()
+            return fig
+        ui.input_select("species", "Species", ["Chinstrap", "Adelie", "Gentoo"])
+```
+</details>
+
+The `render.plot` function above has a lot going on, and so it's easy to overlook that there's a `ui.input_select()` which appears just below the plot, inside the same card.
+Also, with multiple levels of indentation, it's easy to mess up the identation; and as a result, have the input appear in the wrong place, or not work at all.
+
+Express does offer a way to workaround this problem: the `ui.hold()` context manager, which allows you to define a `render` function in one place, then display it in another.
```

Comment:
I think we need to include the Core syntax and why it's good here. Right now it reads:

1) Express has problem
2) Here's how to solve it in Express

I think the idea of this article is that it should motivate Shiny Core, so at the end the reader should know about Core syntax and want to give it a try. Could we do something like break the programming-with-express code into its own article and link to that?